packages/admin/admin-date-time/src/DateTimePicker.tsx

```diff
@@ -0,0 +1,145 @@
+import { Calendar, Time } from "@comet/admin-icons";
+import { ComponentsOverrides, FormControl, InputAdornment, Theme } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import * as React from "react";
+
+import { DatePicker, DatePickerProps } from "./DatePicker";
+import { getDateValueWithTime, getTimeStringFromDate } from "./helpers/timePickerHelpers";
+import { TimePicker, TimePickerProps } from "./TimePicker";
+
+export interface DateTimePickerComponentsProps {
+    datePicker?: Omit<DatePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef" | "fullWidth">;
+    timePicker?: Omit<TimePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef" | "fullWidth">;
+}
+
+export interface DateTimePickerProps {
+    onChange?: (value?: Date) => void;
+    value?: Date;
+    clearable?: boolean;
+    componentsProps?: DateTimePickerComponentsProps;
+}
+
+function DateTimePicker({
+    classes,
+    onChange,
+    value,
+    componentsProps = {},
+    ...propsForBothTimePickers
+}: DateTimePickerProps & WithStyles<typeof styles>) {
+    const datePickerRef = React.useRef<HTMLElement>(null);
+    const timePickerRef = React.useRef<HTMLElement>(null);
+
+    const [valueWasUndefinedBeforeLastOnChange, setValueWasUndefinedBeforeLastOnChange] = React.useState(!value);
+
+    const onChangeDateOrTime = (val?: Date | string) => {
+        if (val === undefined) {
+            onChange?.(undefined);
+            setValueWasUndefinedBeforeLastOnChange(true);
+        } else {
+            setValueWasUndefinedBeforeLastOnChange(false);
+        }
+    };
+
+    const onChangeDate = (date?: Date) => {
+        onChangeDateOrTime(date);
+
+        if (date !== undefined) {
+            const dateValueWithTime = value ? value : new Date();
+            const newDateValue = getDateValueWithTime(date, getTimeStringFromDate(dateValueWithTime));
```

Comment:
The naming seems a bit off. First, you have a variable called `dateValueWithTime`, then you use it as part of the input of a method called `getDateValueWithTime`. 

```suggestion
            const previousDateValueWithTime = value ? value : new Date();
            const newDateValue = getDateValueWithTime(date, getTimeStringFromDate(previousDateValueWithTime));
```